kratos/includes/amatrix_interface.h

```diff
@@ -609,6 +609,19 @@ typename TExpression1Type::data_type inner_prod(
 
 }
 
+    template <typename TExpressionType, std::size_t TCategory>
+    typename TExpressionType::data_type norm_1(
+        AMatrix::MatrixExpression<TExpressionType, TCategory> const& TheExpression) {
+            using data_type = typename TExpressionType::data_type;
+            auto& the_expression = TheExpression.expression();
```

Comment:
why do you need this? I would say you can directly use "TheExpression"

Reply:
yeah. I fixed that, also the std::abs. I closed this one because i did not merge the previous commit locally yet. Has to do that to ensure correct compilation.